backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/mapper/converter/SpecificationConverter.kt

```diff
@@ -34,7 +36,19 @@ class SpecificationConverter constructor(
         }
 
         val content = if (isGraphQLFile) fileContent else specificationDataService.parseFileContent(fileContent)
-        val uuid = specificationFileDto.id ?: UUID.randomUUID()
+
+        val userDefinedId = specificationDataService.readId(content)
+        val urlPathId = specificationFileDto.id
+
+        if (userDefinedId?.let { userId -> !isAcceptable(userId) } ?: run { false } ) {
```

Comment:
Is this better readable than `if(userDefinedId != null && !isAcceptable(userDefinedId))`?